R/complete_gene_expression_heatmap_data.R

```diff
@@ -1,22 +1,56 @@
-complete_gene_expression_heatmap_data <- function(data, input_gene, input_mouse_model) {
-  data %>%
+complete_gene_expression_heatmap_data <- function(data, input) {
+
+  tissues <- magora::gene_expressions_tissue %>%
+    unlist() %>%
+    unique() %>%
+    sort()
+
+  data <- data %>%
     # Ensure genes selected show in plot, even if there is no data
     dplyr::mutate(
       gene = forcats::fct_drop(.data$gene),
-      gene = forcats::fct_expand(.data$gene, input_gene),
-      gene = forcats::fct_relevel(.data$gene, input_gene),
-      gene = forcats::fct_rev(.data$gene)
+      gene = forcats::fct_expand(.data$gene, input$gene),
+      gene = forcats::fct_relevel(.data$gene, input$gene)
     ) %>%
     # Same with mouse_model
     dplyr::mutate(
       mouse_model = forcats::fct_drop(.data$mouse_model),
-      mouse_model = forcats::fct_expand(.data$mouse_model, input_mouse_model),
-      mouse_model = forcats::fct_relevel(.data$mouse_model, input_mouse_model)
+      mouse_model = forcats::fct_expand(.data$mouse_model, input$mouse_model),
+      mouse_model = forcats::fct_relevel(.data$mouse_model, input$mouse_model)
+    ) %>%
+    # And tissue
+    dplyr::mutate(
+      tissue = forcats::fct_drop(.data$tissue),
+      tissue = forcats::fct_expand(.data$tissue, tissues),
+      tissue = forcats::fct_relevel(.data$tissue, tissues)
+    ) %>%
+    # And sex
+    dplyr::mutate(
+      sex = forcats::fct_drop(.data$sex),
+      sex = forcats::fct_expand(.data$sex, input$sex),
+      sex = forcats::fct_relevel(.data$sex, input$sex)
+    ) %>%
+    # And age
+    dplyr::mutate(
+      age_num = .data$age,
+      age = as.character(.data$age),
+      age = forcats::fct_drop(.data$age),
+      age = forcats::fct_expand(.data$age, input$age),
+      age = forcats::fct_relevel(.data$age, input$age)
     ) %>%
     # "Complete" the data set so there is a white square for every combination
     tidyr::complete(.data$mouse_model, .data$tissue, .data$gene, .data$sex, .data$age) %>%
     # Derive a combined sex and age field
-    dplyr::mutate(sex_age = paste(.data$sex, .data$age)) %>%
-    dplyr::arrange(.data$sex, .data$age) %>%
+    dplyr::mutate(sex_age = glue::glue("{.data$sex} ({.data$age} Months)")) %>%
+    dplyr::arrange(.data$sex, .data$age_num) %>%
     dplyr::mutate(sex_age = forcats::fct_inorder(.data$sex_age))
+
+  # Split a long mouse model onto two lines if it's present
+
+  if ("APOE4TREM2" %in% input$mouse_model) {
+    data <- data %>%
+      dplyr::mutate(mouse_model = forcats::fct_recode(.data$mouse_model, `APOE4\nTREM2` = "APOE4TREM2"))
```

Comment:
There's just one case for these long mouse models?

Reply:
Yeah, just one! The rest are 5-6 characters and fit in one line pretty well, so just wanted to split this one (and it seems like rewriting "APOE4TREM2" to "APOE4 TREM2" is a decent way to split the model... scientifically? Like it's not in the real "middle" of a word 😅 )